### PR TITLE
Bump opentelemetry-cpp version to 1.9.1, change LogExporter/LogProcessor to post-1.6 LogRecordExporter and LogRecordProcessor

### PR DESCRIPTION
**Pull request by niande-xbox**
**What?**
The fluentd exporter lacks breaking changes around the naming of Log* classes (which changed to LogRecord* classes).

Remove -DWITH_fluentd option that seems deprecated now.

**What will this enable?**
This should enable adding opentelemetry-fluentd to vcpkg, since the opentelemetry-fluentd port depends on opentelemetry, so we need to be compilation-compatible with the latest version, which is 1.9.1.

**What validation have I run?**
I've been able to see my changes in my own fluentd server:

023-06-20 19:46:35.296130644 +0000 Log: {"ObservedTimestamp":1687290395,"Timestamp":null,"body":"f2","severityNumber":5,"severityText":"DEBUG"}
2023-06-20 19:46:35.296328136 +0000 Log: {"ObservedTimestamp":1687290395,"Timestamp":null,"body":"f3","severityNumber":5,"severityText":"DEBUG"}

I've also passed the gtests:
image

I also repeated the build and gtest-suite on a fresh Ubuntu-20.04 VM.

### CODE REVIEW

File: exporters/fluentd/src/log/recordable.cc [hunk 1]
```diff
@@ -64,6 +72,18 @@ void Recordable::SetTimestamp(
...
+    opentelemetry::common::SystemTimestamp timestamp) noexcept {
+  json_["ObservedTimestamp"] = fluentd_common::get_msgpack_eventtimeext(
+      static_cast<int32_t>(std::chrono::duration_cast<std::chrono::seconds>(
+                               timestamp.time_since_epoch())
+                               .count()),
+      std::chrono::duration_cast<std::chrono::nanoseconds>(
+          timestamp.time_since_epoch())
+              .count() %
+          1000000000);
+}
```
Comment (lalitb) [inline exporters/fluentd/src/log/recordable.cc:85]: should we move this code to some common function to be used in both SetTimestamp and SetObservedTimestamp ?

**Review by lalitb — APPROVED** [review]
Looks Good. We may have to add some versioning for this exporter, and maintain comtability matrix among versions of otel-cpp and fluentd, but can be done outside of this PR. Thanks for contributing.

File: exporters/fluentd/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ Increment the:
 * MINOR version when you add functionality in a backwards compatible manner, and
 * PATCH version when you make backwards compatible bug fixes.
 
+## [2.2.0] 2023-06-20
```
Comment (ThomsonTan) [inline exporters/fluentd/CHANGELOG.md:16]: How is the new version number created?

File: exporters/fluentd/cmake/opentelemetry-cpp-fluentd-config.cmake.in [hunk 3]
```diff
@@ -34,6 +34,7 @@ set(OPENTELEMETRY_CPP_FLUENTD_VERSION
 
 @PACKAGE_INIT@
 
+
```
Comment (ThomsonTan) [inline exporters/fluentd/cmake/opentelemetry-cpp-fluentd-config.cmake.in:37]: nit: remove this empty line.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**